src/components/AnnotationViewer.vue

```diff
@@ -292,6 +311,82 @@ export default class AnnotationViewer extends Vue {
       this.drawNewConnections(displayedIds);
     }
     this.annotationLayer.draw();
+    this.drawTooltips();
+  }
+
+  drawTooltips() {
+    const oldFeatures = this.textLayer.features();
+
+    if (this.shouldDrawTooltips) {
+      // Avoid checking the store for each annotation
+      const all = this.tooltipOnAll;
+      const selected = this.tooltipOnSelected;
+      const hovered = this.tooltipOnHovered;
+
+      const displayedAnnotations = this.annotationLayer
+        .annotations()
+        .map((a: any) => a.options("storedAnnotation"))
+        .filter((a: any) => {
+          return (
+            a &&
+            (all ||
+              (hovered && this.hoveredAnnotationId === a.id) ||
+              (selected && this.isAnnotationSelected(a.id)))
+          );
+        });
+
+      // One text feature per line as in https://opengeoscience.github.io/geojs/tutorials/text/
+      // Centroid is computed once per new line (optimization needed?)
+      // TODO: More performance with renderThreshold https://opengeoscience.github.io/geojs/apidocs/geo.textFeature.html#.styleSpec
+      const baseStyle = {
+        fontSize: "12px",
+        fontFamily: "sans-serif",
+        textAlign: "center",
+        textBaseline: "middle",
+        color: "black",
+        textStrokeColor: "#FFF8",
+        textStrokeWidth: 2
+      };
+      this.textLayer
+        .createFeature("text")
+        .data(displayedAnnotations)
+        .position((annotation: IAnnotation) => {
+          return simpleCentroid(annotation.coordinates);
```

Comment:
Depending on the annotationType, you don't need to compute the centroid everytime.
For pointAnnotation, you can directly set position with `annotation.coordinates[0]`

Reply:
It would be redundant as this is what is done by simpleCentroid

Reply:
Yes, my bad, I was thinking it was checking.
Do you think it is worth it to check for pointAnnotation?
We could put this check in simpleCentroid as it would factorize the code and make it clearer.

Reply:
yes, it avoids doing some computations when it is not needed.
I don't know if it would be clearer, because do we need to compute a centroid for a point, I don't know.
I think, both solution are ok.